src/activity-logger/background/log-page-visit.js

```diff
@@ -52,17 +59,23 @@ export async function logPageVisit({ tabId, url }) {
     // The time to put in documents.
     const timestamp = Date.now()
 
-    // First create an identifier for the page being visited.
-    const storePageResult = await storePage({ tabId, url })
+    const pageId = generatePageDocId({ url })
+    const existingPage = index.initSingleLookup()(pageId)
+
+    let storePageResult
+    if (existingPage == null || timestamp - existingPage.latest < threshold) {
+        // First create an identifier for the page being visited.
+        storePageResult = await storePage({ tabId, url })
+    }
 
     // Create a visit pointing to this page (analysing/storing it may still be in progress)
     const { visit } = await storeVisit({
-        page: storePageResult.page,
+        page: { _id: pageId },
         url,
         timestamp,
     })
 
-    await updateIndex(storePageResult.finalPagePromise, visit)
```

Comment:
Any specific reason to remove the `await` here? It will change the expected behaviour of any callers

Reply:
Oh, this was removed by mistake.